src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts

```diff
@@ -11,25 +11,26 @@ import {
     SetWatermark,
 } from "./../../../../api/kamu.graphql.interface";
 import { SupportedEvents } from "./supported.events";
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from "@angular/core";
+import { ChangeDetectionStrategy, Component, Input } from "@angular/core";
 import { MetadataBlockFragment } from "src/app/api/kamu.graphql.interface";
-import { BaseComponent } from "src/app/common/base.component";
 import { BlockService } from "../../block.service";
 import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { Observable, tap } from "rxjs";
 
 @Component({
     selector: "app-event-details",
     templateUrl: "./event-details.component.html",
     styleUrls: ["./event-details.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class EventDetailsComponent extends BaseComponent implements OnInit {
+export class EventDetailsComponent {
     public block: MetadataBlockFragment;
+    public block$: Observable<MetadataBlockFragment> = this.blockService.onMetadataBlockChanges.pipe(
+        tap((block: MetadataBlockFragment) => (this.block = block)),
+    );
```

Comment:
I suggest moving this to the constructor, for readability and better structuring,
I mean assigning the Observable.

My thoughts:
We will have a section to create parameters,
and we will have a section with initialization and some settings

Reply:
Ok